terraform/deployments/AWS/main.tf

```diff
@@ -57,14 +58,16 @@ module "Kubernetes" {
 
 // Setup host within Kubernetes subnet
 module "InternalNode" {
-  source              = "../../modules/AWS/InternalNode"
-  ami_id              = "${module.Ami.AmiId}"
-  instance_type       = "${var.instance_type}"
-  access_key_name     = "${module.SshKey.KeyPairName}"
-  control_plane_sg_id = "${module.SecurityGroups.ControlPlaneSecurityGroupID}"
-  private_subnet_id   = "${module.Networking.PrivateSubnetId}"
-  default_tags        = "${local.aws_tags}"
-  bastion_public_ip   = "${module.Bastion.BastionPublicIp}"
+  source                   = "../../modules/AWS/InternalNode"
+  ami_id                   = "${module.Ami.AmiId}"
+  instance_type            = "${var.instance_type}"
+  access_key_name          = "${module.SshKey.KeyPairName}"
+  control_plane_sg_id      = "${module.SecurityGroups.ControlPlaneSecurityGroupID}"
+  private_subnet_id        = "${module.Networking.PrivateSubnetId}"
+  default_tags             = "${local.aws_tags}"
+  bastion_public_ip        = "${module.Bastion.BastionPublicIp}"
+  iam_instance_profile_id  = "${module.S3Storage.IamInstanceProfileId}"
```

Comment:
Is this the right profile?

Reply:
Yes as it's the profile the nodes and master use to push and pull from S3. I'm not sure it's worth it to create a new profile just for the internal instance.

Reply:
I would rather have separate profiles to be honest.

Reply:
> I would rather have separate profiles to be honest.

are you sure,it would just be a duplicate of the role to access S3?